userguide/02. Intro to Artifacts.ipynb

```diff
@@ -0,0 +1,404 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from __future__ import absolute_import\n",
+    "from __future__ import print_function\n",
+    "\n",
+    "from tornado import gen\n",
+    "from flowz import app\n",
+    "from flowz.channels import *\n",
+    "\n",
+    "from flowz.artifacts import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "from logging import config as logconf\n",
+    "def config_logging(level='DEBUG'):\n",
+    "    logconf.dictConfig({\n",
+    "            'version': 1,\n",
+    "            'loggers': {},\n",
+    "            'disable_existing_loggers': 0,\n",
+    "            'root': {\n",
+    "                'level': level,\n",
+    "                'handlers': ['default_handler'],\n",
+    "                },\n",
+    "            'handlers': {\n",
+    "                'default_handler': {\n",
+    "                    'class': 'logging.StreamHandler',\n",
+    "                    'stream': 'ext://sys.stdout',\n",
+    "                    },\n",
+    "                },\n",
+    "            }\n",
+    "    )\n",
+    "config_logging('DEBUG')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def print_chans(*chans):\n",
+    "    # This is a bit more elaborate than before to resolve artifacts\n",
+    "    app.Flo([chan.map(lambda y: getattr(y, 'get')()).each_ready().map(print) for chan in chans]).run()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Introduction to Artifacts"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Background"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If channels only used sources like lists and dictionaries already in memory, they would have little value. The true values comes when getting data from external sources or heavy computations that take time.  In such an environment, accessing the data asynchronously and concurrently -- and even \"out of order\" in some cases -- can lead to nice performance benefits, and possibly more elegant code.\n",
+    "\n",
+    "The key to unlocking that asynchrony is artifacts. Artifacts are objects that know how to retrieve, compute, or transform their data, but don't necessarily do it right away.  They delay getting the data until requested to do so, and then they use the tornado/futures infrastructure to get their data asynchronously.  Once ready, their data is available to others."
```

Comment:
Yes, you could do that. The rest of us will probably use artifacts. :)

I've softened the assertion.